### PR TITLE
refactor: basic example from class component to functional component

### PR DESCRIPTION
**Pull request by seyedmostafahasani**
## Summary
Refactor basic example from class component to functional component.

### Motivation
Fix some TypeScript issues and improve the readability of the functional component compared to the class component.

## Test plan

### CODE REVIEW

File: examples/basic/src/utils/Toast.ts [hunk 1]
```diff
@@ -0,0 +1,18 @@
+import {Alert, ToastAndroid} from 'react-native';
```
Comment (freeboub) [inline examples/basic/src/utils/Toast.ts]: I think this file should go into components folder

**Review by freeboub — CHANGES_REQUESTED** [review]
I added a small comment about Toast file but except this, changes looks good to me.
In a more general consideration:
- I don't think the split of overlay really bring interesting thing, but it goes in the good way
- please try to do smaller PR, it is really easier to review
- the global state also need to be removed/reworked. but let's keep it for later, there are enough changes in this PR

Please move Toast.ts and I will approve

**Comment by seyedmostafahasani** [comment]
Sorry about the big PR. I wanted to do it in some small PRs, but all of them were related to each other, so I decided to combine them into this PR. I agree with you about the benefits of small PRs 🤝.

**Review by freeboub — APPROVED** [review]

**Review by freeboub — APPROVED** [review]

**Comment by KrzysztofMoch** [comment]
@seyedmostafahasani seekbar is not working for me (on iOS)
And Btw local file (first source in example) is not loading for me, I have warning that I am try to load empty source - It's me or example broke somewhere ?

**Comment by freeboub** [comment]
@KrzysztofMoch I think broken local videos issue comes from expo PR... Sometimes it works, sometimes not 🤨

**Comment by seyedmostafahasani** [comment]
@KrzysztofMoch I saw your comments when I was refactoring. I tested on the master branch just to be on the safe side, and I noticed we have these issues. I think like @freeboub. Also seekbar is not working on Android.

**Comment by KrzysztofMoch** [comment]
Seekbar is not working on master ? 👀

**Comment by seyedmostafahasani** [comment]
Yeah. It's not working correctly. Please see the video below.

https://github.com/TheWidlarzGroup/react-native-video/assets/3009224/903a4350-de42-4e05-a68e-2bcca8bf5ba9

**Comment by KrzysztofMoch** [comment]
> Yeah. It's not working correctly. Please see the video below.
> 
>  Screen.Recording.2024-06-27.at.17.17.19.mov

This is actually working - it is just live source. I have issue with normal video, seekbar is not updating it's position (it's staying in one position)

**Comment by seyedmostafahasani** [comment]
Yeah, you're right. Let me check it out on iOS first on the master branch, then I will test it on this PR.

**Comment by seyedmostafahasani** [comment]
@KrzysztofMoch It's fixed. We need to refactor the state variable in another PR in the future.

**Review by KrzysztofMoch — APPROVED** [review]
Thanks, now it's working

**Comment by seyedmostafahasani** [comment]
@KrzysztofMoch thanks for testing again!

**Pull request merged**